enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,310 @@
+---
+title: cluster-logging-log-forwarding
+authors:
+  - "@jcantrill"
+reviewers:
+  - "@bparees"
+  - "@ewolinetz"
+  - "@jeremyeder"
+approvers:
+  - TBD
+creation-date: 2019-09-17
+last-updated: 2019-09-17
+status: implementable
+see-also:[]
+replaces:[]
+superseded-by:[]
+---
+
+# cluster-logging-log-forwarding
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The purpose of log forwarding is to provide a declarative way by which adopters of cluster logging can ship
+container and node logs to destinations that are not necessarily managed by the OKD cluster logging infrastructure.
+Destinations are either on or off cluster endpoints such as the cluster logging provided Elasticsearch, an organization's Kafka message bus, or other log aggregation solutions (e.g. Splunk, Graylog).
+
+## Motivation
+
+Organizations desire to reuse their existing enterprise log solutions to store container logs.  Providing a declarative mechanism by which administrators define a log destination simplifies their operational burden.  The cluster logging stack is able to deploy a collector to each OKD node that is configured with the necessary permissions to collect the logs, add container metadata (e.g. labels) and ship them to the specified endpoint.
+
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that identifies log sources and their desired destination
+* Send logs to an Elasticsearch cluster not managed by the cluster logging infrastructure
+* Send logs to a fluentd instance not managed by the cluster logging infrastructure via fluentd's secureforward protocol
+* Allow deployment of log forwarding without deploying the entirety of the cluster logging infrastructure (e.g. Kibana, Elasticsearch)
+* Support TLS between the collector and destination if so configured
+* Specifically only forward "application" and "infrastructure" logs
+* Re-use log forwarding to write logs to a cluster logging managed log aggregator (e.g. Elasticsearch)
```

Comment:
will forwarding to the "managed" elastic cluster be configured the same way as any other target?  or is it treated" specially" by the forwarding configuration/logic?

Reply:
Treated the same

Reply:
probably worth stating this general premise explicitly.

also what does that mean for how the user provided config bits(ones forwarding logs elsewhere) interact with what i assume will be operator generated config bits (the ones wiring up our internal collection/logstore)?  How do we keep them from changing our settings or what do we do if they do?

Reply:
> also what does that mean for how the user provided config bits(ones forwarding logs elsewhere) interact with what i assume will be operator generated config bits (the ones wiring up our internal collection/logstore)? How do we keep them from changing our settings or what do we do if they do?

When writing to CLO managed store, there are no config bits or concerns that are relevant to users because the cert gen and regen are managed by CLO.  We state that users need to create a secret with the appropriate certs that we consume in order to write to a service provided by users.  It is a user's responsibility to change those certs when needed.  Is this different then any other user managed service?  Are there examples of other cluster services that take ownership of secrets they did not create?

Reply:
if there are no config bits for the internal store then i guess that is fine.